bgp/modules/terms.py

```diff
@@ -113,3 +115,45 @@ def validate_isbn(isbn):
     def __init__(self):
         super(IsbnExtractorModule, self).__init__(self.validate_isbn)
 
+
+class PageTypeProcessor(object):
+
+    def __init__(self, modules,keyword):
+        """
+        :param modules: a dict of {'name': module}
+        """
+        self.modules = modules
+        self.keyword = keyword
+    
+    def run(self, book):
+        self.pageNo = self.search(book.xml,keyword = self.keyword)
+
+    @classmethod
+    def search(cls,xmlFile,keyword = ""):
```

Comment:
Rather than search, the PageTypeProcessor should probably process one **page at a time** and pass each page to some function on/of its modules.

e.g. "I'm a PageTypeProcessor. My job is to iterate over each xml page, pass the page to each of my modules (e.g. TableofContentsPageDetectorModule, KeywordPageDetectorModule, TitlePageDetectorModule) and should the results should be added/accumulated/stored within the PageTypeProcessor.

Reply:
PageTypeProcessor is the parent class. So are we trying to call the function of child modules from parent class? 
Doing this seems a little odd.